### PR TITLE
Change ownership of PCF to platform integrations

### PR DESCRIPTION
**Pull request by hithwen**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -112,7 +112,7 @@
 /cmd/cluster-agent/                     @DataDog/container-integrations
 /cmd/cluster-agent/commands/            @DataDog/container-integrations @DataDog/integrations-tools-and-libraries
```
Comment (hithwen) [inline .github/CODEOWNERS:113]: not sure about this one

**Review by djmitche — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (NouemanKHAL) [reply .github/CODEOWNERS:113]: I don't think we own anything on that folder, it's probably outdated.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -112,7 +112,7 @@
...
 /cmd/cluster-agent-cloudfoundry/        @DataDog/integrations-tools-and-libraries
```
Comment (NouemanKHAL) [inline .github/CODEOWNERS:114]:
```suggestion
/cmd/cluster-agent-cloudfoundry/        @DataDog/platform-integrations 
```

**Comment by NouemanKHAL** [comment]
We should also update `.gitlab/JOBOWNERS` and `tasks/pipeline.py` (for the `GITHUB_SLACK_MAP`)

**Review by mx-psi — APPROVED** [review]

**Review by therve — APPROVED** [review]

**Pull request merged**